db/record.c

```diff
@@ -2895,6 +2887,21 @@ static int check_blob_buffers(struct ireq *iq, blob_buffer_t *blobs,
     return 0;
 }
 
+static int check_blob_sizes(struct ireq *iq, blob_buffer_t *blobs, int maxblobs) {
+    int retrc = 0;
+
+    for (int i = 0; i < maxblobs; i++) {
+        if (blobs[i].exists && blobs[i].length > MAXBLOBLENGTH) {
+            reqerrstr(iq, COMDB2_ADD_RC_INVL_BLOB,
+                    "blob size (%d) exceeds maximum (%d)",
+                    blobs[i].length, MAXBLOBLENGTH);
+            retrc = ERR_BLOB_TOO_LARGE;
```

Comment:
On error, should be ok to return directly from here (no need for retrc). I'd also move this definition at the place of declaration.